app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -24,9 +24,15 @@ def callback_for(provider)
     else
       session["devise.#{provider}_data"] = request.env["omniauth.auth"]
       user_errors = @user.errors.full_messages
+      logger.error "Log in error: sign in failed. User username and email: " + @user.username + " - " + @user.email
+      logger.error "Log in error: auth data hash - " + request.env["omniauth.auth"].to_s
```

Comment:
This might be dangerous because it would log their OAuth tokens. Not sure what other data to log though.